host_source.go

```diff
@@ -96,6 +96,11 @@ func (r *ringDescriber) GetHosts() (hosts []HostInfo, partitioner string, err er
 		localHost.Peer = addr
 	}
 
+	// Make sure we didn't screw up discovering our local peer.
+	if localHost.Peer == "" || localHost.Peer == "0.0.0.0" {
+		return nil, "", ErrHostQueryFailed
```

Comment:
this was added as a safety check, but it could be removed. It was 0.0.0.0 only when selecting on rpc_address in previous version.